### PR TITLE
New usage rights model

### PR DESCRIPTION
**Pull request by jamesgorrie**
# New :sun_with_face: 
* One source of truth - [UsageRights](https://github.com/guardian/grid/blob/new-usage-rights-model/common-lib/src/main/scala/com/gu/mediaservice/model/UsageRights.scala)
* Json parsing is all in one place and will act the same throughout the system
* Modelled more closely to the idea of having 1 type of usage rights, e.g. Agency.
* Validation through case class instantiation

## Before launch
- [ ] Update mappings

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala [hunk 1]
```diff
@@ -154,10 +154,13 @@ object AddAgencyCategory extends ImageProcessor {
...
-    if (shouldHaveAgency(image)) imageWithAgency(image) else image
+    isAgency(image).map(supplier => imageWithAgency(image, supplier)).getOrElse(image)
 
-  def imageWithAgency(image: Image): Image =
-    image.copy(usageRights = image.usageRights.copy(category = Some(Agency)))
+  def imageWithAgency(image: Image, supplier: String): Image =
+    image.copy(usageRights = Agency(supplier))
 
-  def shouldHaveAgency(image: Image): Boolean = image.usageRights.supplier.nonEmpty
+  def isAgency(image: Image): Option[String] = image.usageRights match {
```
Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala]: Removed as redundant.

**Comment by jamesgorrie** [comment]
@theefer bump

**Comment by kenoir** [comment]
@jamesgorrie the UI errors with `TypeError: Cannot read property 'toLocaleString' of undefined` when trying to run this branch locally.

**Comment by jamesgorrie** [comment]
That's normally when your API is down and us not handling the error very well.

**Comment by kenoir** [comment]
My fault, media-api broke locally for some reason. Looks fine to me :+1:

**Pull request merged**